### PR TITLE
Implement profiling for rules and functions

### PR DESCRIPTION
**Pull request by mmussomele**

### CODE REVIEW

File: repl/repl.go [hunk 1]
```diff
@@ -1170,6 +1213,7 @@ var builtin = [...]commandDesc{
 	{"json", []string{}, "set output format to JSON"},
 	{"pretty", []string{}, "set output format to pretty"},
 	{"trace", []string{}, "toggle full trace"},
+	{"metrics", []string{}, "toggle profiling"},
```
Comment (tsandall) [inline repl/repl.go]: Nit: `toggle metrics`.

Comment (tsandall) [inline repl/repl.go]: The `metrics` command would be great! Please pull that out into a separate PR. We can probably merge it right away.

Comment (mmussomele) [inline repl/repl.go]: so for now, leave metrics (parse, compile, eval) in the REPL, but back off on profiling until we get the details right?

Comment (tsandall) [inline repl/repl.go]: Yes, that would be my preference.

**Comment by mmussomele** [comment]
Closing in favor of #410
